iothub_client/src/message_queue.c

```diff
@@ -54,7 +54,15 @@ static bool find_item_by_message_ptr(LIST_ITEM_HANDLE list_item, const void* mat
     MESSAGE_QUEUE_ITEM* current_item = (MESSAGE_QUEUE_ITEM*)singlylinkedlist_item_get_value(list_item);
     MQ_MESSAGE_HANDLE* target_item = (MQ_MESSAGE_HANDLE*)match_context;
 
-    return (current_item->message == target_item);
+    if (current_item == NULL)
```

Comment:
the style in this doc (and I think most of this SDK) is to have one return point. So please update to have something like 
```c
bool result;

if()
{
    result = false;
}
else
{
//...
}

return result;
```

Reply:
else:
result = (current_item->message == target_item);